pulp_container/app/serializers.py

```diff
@@ -230,6 +241,25 @@ def validate(self, data):
                         "repository version."
                     )
                 )
+        base_path = validated_data["base_path"]
+        namespace_name = base_path.split("/")[0]
+
+        if "namespace" in validated_data:
+            if validated_data["namespace"].name != namespace_name:
+                raise serializers.ValidationError(
+                    _("Selected namespace does not match first component of base_path.")
+                )
+        else:
+            try:
+                validated_data["namespace"] = models.ContainerNamespace.objects.get(
+                    name=namespace_name
+                )
+            except ObjectDoesNotExist:
+                namespace_serializer = ContainerNamespaceSerializer(data={"name": namespace_name})
+                namespace_serializer.is_valid(raise_exception=True)
+                validated_data["namespace"] = namespace_serializer.create(
+                    namespace_serializer.validated_data
+                )
```

Comment:
there's room here for a race condition, if two repos are being created with a new namespace at the same time it could result in one of the 'creates' to fail when the other task created it.  Not sure how often ya'll deal with that within pulp, but we see it all the time within katello

Reply:
Sounds like we really need a sophisticated get_or_create on the serializer model.
I think we do that in >=3 places.

Reply:
I added another except block to recognize a lost race.